File: Jardin De infantes/UI/Servicios/Idioma/iuIdioma.vb
Imports BLL
Imports EE

Imports System.Globalization
Imports System.Threading

Public Class iuIdioma

    Sub cambiaridioma()

        Dim unbllTraducir As New bllTraduccion


        Dim unTraducir As New Traduccion(SingletonIdioma.intance.getUsuario)
        Dim ht As New Hashtable
        ht = unbllTraducir.ListarPalabrasTraducidas(unTraducir)
        For Each item As DictionaryEntry In ht
            For Each controlcito As Control In Me.Controls
                If controlcito.Tag Is Nothing Then
                Else
                    If controlcito.Tag.Equals(item.Key) Then
                        controlcito.Text = item.Value

                    End If
                End If





            Next
        Next
        Dim culturaObj As CultureInfo = New CultureInfo(SingletonIdioma.intance.getUsuario.getIdioma)
        Threading.Thread.CurrentThread.CurrentUICulture = culturaObj
        Threading.Thread.CurrentThread.CurrentCulture = culturaObj
    End Sub
    Private Sub mtCambiarIdioma_Click(sender As Object, e As EventArgs) Handles mtCambiarIdioma.Click
        Dim unIdioma As New iuCambiarIdioma
        unIdioma.ShowDialog()
        cambiaridioma()


    End Sub

    Private Sub mtAgregarIdioma_Click(sender As Object, e As EventArgs) Handles mtAgregarIdioma.Click
        Dim frm As New iuAltaIdioma
        frm.ShowDialog()

    End Sub

    Private Sub iuIdioma_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        cambiaridioma()

    End Sub
End Class

File: Jardin De infantes/Seguridad/IU/iuIdioma.cs
using BLLseguridad.Idioma;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace IU
{
    public partial class iuIdioma : MetroFramework.Forms.MetroForm
    {

        void CambiarCultura()
        {
            CultureInfo cultura = new CultureInfo(EEseguridad.Idioma.SingletonIdioma.intance.getUsuario().idioma);
            System.Threading.Thread.CurrentThread.CurrentUICulture = cultura;
            System.Threading.Thread.CurrentThread.CurrentCulture = cultura;

        }
        void CambiarIdioma()
        {
            bllTraduccion bll = new bllTraduccion();
            


            
            bllTraduccion traduccion = new bllTraduccion();
            Hashtable ht = new Hashtable();
            ht = traduccion.ObtenerTraduccion(EEseguridad.Idioma.SingletonIdioma.intance.getUsuario().idioma);
            foreach (DictionaryEntry item in ht)
            {
                foreach (Control controles in this.Controls)
                {
                    if (controles.Tag==null)
                    {

                    }
                    else
                    {
                        if (controles.Tag.Equals(item.Key))
                        {
                            controles.Text = item.Value.ToString();
                        }
                    }
                }
            }


        }
        public iuIdioma()
        {
            InitializeComponent();
        }

        private void iuIdioma_Load(object sender, EventArgs e)
        {
            helpProvider1.HelpNamespace = CHM.Cadena;

            CambiarCultura();
            CambiarIdioma();


        }

        private void mtCambiarIdioma_Click(object sender, EventArgs e)
        {
            iuCambiarIdioma iu = new iuCambiarIdioma();
            iu.ShowDialog();
            CambiarCultura();
            CambiarIdioma();
        }

        private void mtAgregarIdioma_Click(object sender, EventArgs e)
        {
            iuCrearIdioma idioma = new iuCrearIdioma();
            idioma.ShowDialog();
        }
    }
}
